### PR TITLE
Capture Future errors for Flutter Web automatically

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Capture Future errors for Flutter Web automatically

## :bulb: Motivation and Context
https://github.com/flutter/flutter/issues/100277#issuecomment-1327478616

## :green_heart: How did you test it?
Running Flutter Web and non-Web, throwing `Future` errors, and checking the error handlers.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: flutter/lib/src/integrations/flutter_error_integration.dart [hunk 1]
```diff
@@ -60,11 +60,6 @@ class FlutterErrorIntegration extends Integration<SentryFlutterOptions> {
         final mechanism = Mechanism(
           type: 'FlutterError',
           handled: true,
-          data: {
```
Comment (marandaneto) [inline flutter/lib/src/integrations/flutter_error_integration.dart:63]: Not really needed anymore, people are used to scrolling down and reading all the available context.

**Comment by marandaneto** [comment]
@ueman are you aware of this problem?
@denrase do you remember if you tested the new error handler on Web?

**Comment by marandaneto** [comment]
@krystofwoldrich or @brustolin would love a review here.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**